lib/opencensus/trace/truncatable_string.rb

```diff
@@ -0,0 +1,52 @@
+# Copyright 2017 OpenCensus Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module OpenCensus
+  module Trace
+    ##
+    # A string that might be shortened to a specified length.
+    #
+    class TruncatableString
+      ##
+      # The shortened string. For example, if the original string was 500 bytes
+      # long and the limit of the string was 128 bytes, then this value contains
+      # the first 128 bytes of the 500-byte string. Note that truncation always
+      # happens on a character boundary, to ensure that a truncated string is
+      # still valid UTF-8. Because it may contain multi-byte characters, the
+      # size of the truncated string may be less than the truncation limit.
+      #
+      # @return [String]
+      #
+      attr_reader :value
```

Comment:
Define `to_s` also, for convenience.

Reply:
Explicitly defined a `to_s` method rather than aliasing the method.